### ISSUE TITLE
v7 to v8  -  no legends shown for several layers

### ISSUE THREAD

**Issue by juan-guerschman** [TerriaJS/geoglam-nm#109]
![image](https://user-images.githubusercontent.com/8801702/107480907-74fce600-6bd1-11eb-9252-5c5c931b2c21.png)

seems to be with GSKY layers, MapServer, 

Geoserver seems ok (but I haven't tested all of them)

**Comment by nf-s** [comment]
Hi @juan-guerschman I think i have fixed the issue:

- The Legends weren't displaying due to us changing how we request WMS legends - the GSKY WMS doesn't advertise legends for those layers or `GetLegendGraphic` support in GetCapabilities. Our new default is that we don't request `GetLegendGraphic` unless it is advertised, or we can identify the service as a Geoserver or NcWMS. To overcome this I have set the `supportsGetLegendGraphic` catalog property to `true`.
- Date formatting was due to catalog configuration - we had ` "dateFormat": "yyyy"` which was setting formatting in the DatePicker - I have removed this.
- I check MapServer legends and they seem to be working - I just upgraded the terriajs-version so it may have been an older bug.

Also we have a new URL https://beta.geo-rapp.terria.io/ - as the old one was pointing directly to our cluster. Let me know if this works for you!
